sgt-back-end/index.js

```diff
@@ -0,0 +1,139 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    }
+    )
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const grade = parseInt(req.body.grade, 10);
+  const name = req.body.name;
+  const course = req.body.course;
+  if (!name || !course || !Number.isInteger(grade) || grade <= 0) {
+    return res.status(400).json({
+      error: 'Submission must contain valid name/course/positive integer!'
+    });
+  }
+  const sql = `
+    insert into
+      "grades" ("name", "course", "grade")
+      values ($1, $2, $3)
+      returning*
+  `;
+  const params = [
+    name,
+    course,
+    grade
+  ];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: 'Cannot add invalid grade'
+        });
```

Comment:
A successful insert will never return a 404. you can remove this conditional logic